mlrun/artifacts/manager.py

```diff
@@ -149,34 +149,36 @@ def log_artifact(
             else:
                 item = Artifact(key, body, **kwargs)
         else:
-            key = item.key
-            target_path = target_path or item.target_path
+            key = item.metadata.key
+            target_path = target_path or item.spec.target_path
 
-        src_path = local_path or item.src_path  # TODO: remove src_path
+        src_path = local_path or item.spec.src_path  # TODO: remove src_path
         if format == "html" or (src_path and pathlib.Path(src_path).suffix == "html"):
             viewer = "web-app"
-        item.format = format or item.format
-        item.src_path = src_path
+        item.spec.format = format or item.spec.format
+        item.spec.src_path = src_path
 
         if db_key is None:
             # set the default artifact db key
             if producer.kind == "run":
                 db_key = producer.name + "_" + key
             else:
                 db_key = key
-        item.db_key = db_key if db_key else ""
-        item.viewer = viewer or item.viewer
-        item.tree = producer.tag
-        item.tag = tag or item.tag
+        item.spec.db_key = db_key if db_key else ""
+        item.spec.viewer = viewer or item.spec.viewer
+        item.metadata.tree = producer.tag
+        item.metadata.tag = tag or item.metadata.tag
```

Comment:
I think it is still possible to work with legacy artifacts which doesn't contain `spec & metadata`. 
Take a look at the new Artifact class as you can see we added methods which are still backwards compatible with the legacy artifact but they actually sets the values into the correct attribute structure ( e.g you set `item.tag` and if it new artifact it will be set to `item.metadata.tag`). 
I think those changes to set the attributes to `metadata / spec` will be done only after we fully deprecate LegacyArtifact and until then we will have to live with those depreciation warnings, or add setters and getters to the Legacy ( which I think is a waste of time to work on it as we will removed in the near future)

Reply:
yes, I will rollback these edits, I edited these as an example of what maybe should be done